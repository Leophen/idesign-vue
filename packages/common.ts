import _ from 'lodash'
import {
  Component,
  getCurrentInstance,
  Slots,
  VNode,
} from 'vue'

// 通用位置类型
export type placementType =
  | 'top'
  | 'top-left'
  | 'top-right'
  | 'bottom'
  | 'bottom-left'
  | 'bottom-right'
  | 'left'
  | 'left-top'
  | 'left-bottom'
  | 'right'
  | 'right-top'
  | 'right-bottom'

// 通用触发类型
export type triggerType = 'hover' | 'click' | 'context-menu'

// 转换 width/height/left/... 样式值
export const turnValue = (val: string | number | undefined) => {
  if (!val) {
    return
  }
  if (_.isNumber(val as number)) {
    return `${val}px`
  }
  return val
}

// const _stateValue = ref(defaultValue)
// const innerValue = computed(() => value ?? _stateValue.value)

/**
 * 渲染 default slot，获取子组件 VNode。处理多种子组件创建场景
 * @returns {function(childComponentName: string, slots: Slots): VNode[]}
 * @param childComponentName
 * @param slots
 * @example const getChildByName = useChildComponentSlots()
 * @example getChildComponentByName('xxItem')
 */
export const useChildComponentSlots = () => {
  const instance = getCurrentInstance()
  return (childComponentName: string, slots?: Slots): VNode[] => {
    if (!slots) {
      slots = instance?.slots
    }
    const content = slots?.default?.() || []

    return (
      content
        .map((item: VNode) => {
          if (item.children && Array.isArray(item.children))
            return item.children
          return item
        })
        .flat()
        // @ts-ignore
        .filter((item: VNode) =>
          (item.type as Component).name?.endsWith(childComponentName)
        ) as VNode[]
    )
  }
}
