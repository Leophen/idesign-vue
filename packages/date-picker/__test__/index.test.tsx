import { DatePicker } from '../index';
import { render } from '../../common'

describe('DatePicker 组件测试', () => {
  it('value', () => {
    const wrapper = render(<DatePicker modelValue='2022-05-20' />);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('defaultValue', () => {
    const wrapper = render(<DatePicker defaultValue='2022-05-20' />);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('disabled', () => {
    const wrapper = render(<DatePicker disabled />);
    expect(wrapper.find('.i-input').classes()).toContain('i-input-is-disabled');
  });

  it('type', () => {
    const wrapper = render(<DatePicker type="range" modelValue={['2022-05-20', '2022-11-11']} />);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('rangeSeparator', () => {
    const wrapper = render(<DatePicker type="range" modelValue={['2022-05-20', '2022-11-11']} rangeSeparator="/" />);
    expect(wrapper.element).toMatchSnapshot();
  });
});

