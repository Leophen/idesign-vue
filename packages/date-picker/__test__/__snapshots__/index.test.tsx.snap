// Vitest Snapshot v1

exports[`DatePicker 组件测试 > defaultValue 1`] = `
<div
  class="i-date-picker"
  data-type="date"
>
  
  <div
    class="i-popup__reference"
  >
    
    
    <div
      class="i-input i-input-is-readonly"
    >
      <!---->
      <!---->
      <input
        class="i-input__inner"
        placeholder="请选择日期"
        readonly=""
      />
      <!---->
      <!---->
      <!---->
      <i
        class="i-icon icon-Calendar i-input-suffix-icon"
      />
      <!---->
      <!---->
    </div>
    <!---->
    
    
  </div>
  <!--teleport start-->
  <!--teleport end-->
  
</div>
`;

exports[`DatePicker 组件测试 > rangeSeparator 1`] = `
<div
  class="i-date-picker"
  data-type="range"
>
  
  <div
    class="i-popup__reference"
  >
    
    
    <div
      class="i-input i-input-is-readonly"
    >
      <!---->
      <!---->
      <input
        class="i-input__inner"
        placeholder="请选择日期"
        readonly=""
      />
      <!---->
      <!---->
      <!---->
      <i
        class="i-icon icon-Calendar i-input-suffix-icon"
      />
      <!---->
      <!---->
    </div>
    <!---->
    
    
  </div>
  <!--teleport start-->
  <!--teleport end-->
  
</div>
`;

exports[`DatePicker 组件测试 > type 1`] = `
<div
  class="i-date-picker"
  data-type="range"
>
  
  <div
    class="i-popup__reference"
  >
    
    
    <div
      class="i-input i-input-is-readonly"
    >
      <!---->
      <!---->
      <input
        class="i-input__inner"
        placeholder="请选择日期"
        readonly=""
      />
      <!---->
      <!---->
      <!---->
      <i
        class="i-icon icon-Calendar i-input-suffix-icon"
      />
      <!---->
      <!---->
    </div>
    <!---->
    
    
  </div>
  <!--teleport start-->
  <!--teleport end-->
  
</div>
`;

exports[`DatePicker 组件测试 > value 1`] = `
<div
  class="i-date-picker"
  data-type="date"
>
  
  <div
    class="i-popup__reference"
  >
    
    
    <div
      class="i-input i-input-is-readonly"
    >
      <!---->
      <!---->
      <input
        class="i-input__inner"
        placeholder="请选择日期"
        readonly=""
      />
      <!---->
      <!---->
      <!---->
      <i
        class="i-icon icon-Calendar i-input-suffix-icon"
      />
      <!---->
      <!---->
    </div>
    <!---->
    
    
  </div>
  <!--teleport start-->
  <!--teleport end-->
  
</div>
`;
